source/data/update.sql

```diff
@@ -13,20 +38,26 @@ CREATE MATERIALIZED VIEW idetect_fact_api AS (
     idetect_fact_locations.location,
     idetect_analyses.gkg_id,
     idetect_analyses.category,
-    idetect_analyses.content_id
+    idetect_analyses.content_id,
+    idetect_fact_api_locations.location_ids_idx
   from idetect_facts
   join idetect_analysis_facts ON idetect_facts.id = idetect_analysis_facts.fact
   join idetect_analyses ON idetect_analysis_facts.analysis = idetect_analyses.gkg_id
   join gkg ON gkg.id = idetect_analyses.gkg_id
   left join idetect_fact_locations ON idetect_facts.id = idetect_fact_locations.fact
+  left join idetect_fact_api_locations ON idetect_fact_api_locations.fact=idetect_facts.id
 );
+ALTER TABLE idetect_fact_api OWNER TO idetect;
 
+CREATE INDEX idetect_fact_api_fact_day_idx on idetect_fact_api (fact,gdelt_day);
 CREATE INDEX idetect_fact_api_loc_day_idx on idetect_fact_api (location, gdelt_day);
 CREATE INDEX idetect_fact_api_day_loc_idx on idetect_fact_api (gdelt_day, location);
-CREATE INDEX idetect_fact_api_loc_idx on idetect_fact_api (location, category);
-CREATE INDEX idetect_fact_api_day_idx on idetect_fact_api (gdelt_day, category);
-CREATE INDEX idetect_fact_api_category_idx on idetect_fact_api (category);
-create INDEX idetect_fact_api_fact_hash ON idetect_fact_api USING HASH (fact);
+CREATE INDEX idetect_fact_api_loc_cat_idx on idetect_fact_api (location, category);
+CREATE INDEX idetect_fact_api_day_cat_idx on idetect_fact_api (gdelt_day, category);
+CREATE INDEX idetect_fact_api_cat_idx on idetect_fact_api (category);
+CREATE INDEX idetect_fact_api_fact_hash ON idetect_fact_api USING HASH (fact);
+CREATE INDEX idetect_fact_api_locations_fact_hash ON idetect_fact_api_locations USING HASH (fact);
```

Comment:
Actually I tried both and HASH is a bit (probably in a negligible way) faster. Since you used HASH for fact in idetect_fact_api I thought it might be a good idea to use the same. I changed it to Btree